src/Umbraco.Web/umbraco.presentation/umbraco/create/MemberGroupTasks.cs

```diff
@@ -1,45 +1,54 @@
-using System;
-using System.Data;
+using System.Linq;
 using System.Web.Security;
-using Umbraco.Web.UI;
 using umbraco.BusinessLogic;
-using umbraco.DataLayer;
-using umbraco.BasePages;
-using Umbraco.Core.IO;
 using umbraco.cms.businesslogic.member;
+using Umbraco.Web.UI;
 
-namespace umbraco
+namespace Umbraco.Web.umbraco.presentation.umbraco.create
 {
     public class MemberGroupTasks : LegacyDialogTask
     {
         public override bool PerformSave()
         {
             Roles.CreateRole(Alias);
-            _returnUrl = string.Format("members/EditMemberGroup.aspx?id={0}", System.Web.HttpContext.Current.Server.UrlEncode(Alias));
+            _returnUrl = $"members/EditMemberGroup.aspx?id={System.Web.HttpContext.Current.Server.UrlEncode(Alias)}";
             return true;
         }
 
         public override bool PerformDelete()
         {
+            var roleDeleted = false;
+
             // only built-in roles can be deleted
             if (Member.IsUsingUmbracoRoles())
             {
-                MemberGroup.GetByName(Alias).delete();
-                return true;
+                Roles.DeleteRole(Alias);
+                roleDeleted = true;

```

Comment:
@RobertCopilau `DeleteRole` returns a boolean, so you can merge these two lines! :-)

Reply:
Done!